src/chart/bar/BarView.js

```diff
@@ -195,6 +247,11 @@ export default echarts.extendChartView({
                 );
             })
             .remove(function (dataIndex) {
+                var bgEl = backgroundGroup.childOfName(dataIndex + '');
+                if (bgEl) {
```

Comment:
Using `childOfName` is inefficient. 

I suggest we always create a new `group` for storing background elements and using `oldBackgroundGroup.childAt(dataIndex)` to get the previously added `bgEl`.

So the code can be.

```js
// Always do remove the previous background group.
// `remove`will handle the `null` value.
var oldBackgroundGroup = group.childOfName('background');
group.remove(oldBackgroundGroup);
var oldLargeBackgroundEl = group.childOfName('largeBackground');
group.remove(oldLargeBackgroundEl);

var newBackgroundGroup = new Group({ name: 'background' });
group.add(newBackgroundGroup);

data.diff(oldData)
  .add(function () {
     newBackgroundGroup.add(bgEl);
   })
  .update(function (newIndex, oldIndex) {
     var bgEl = oldBackgroundGroup.childAt(oldIndex);
     updateProps(bgEl, ....);
     // Add to the new group
     newBackgroundGroup.add(bgEl);
  })
  .remove(function (oldIndex) {
      // We don't need to do anything here because the bgEl don't need to be added to newBackgroundGroup

      // Or if you want to show fadeout animation.
     var bgEl = oldBackgroundGroup.childAt(oldIndex);
      bgEl.animateTo(...., function () {
         // Remove after fade out
         newBackgroundGroup.remove(bgEl);
      });
      newBackgroundGroup.add(bgEl);
   });
```

In this way to manage the background group is more robust and efficient. We don't need to think about when to remove the elements.